test/jdk/java/text/Format/NumberFormat/NumberRegression.java

```diff
@@ -1776,7 +1776,7 @@ public void test4241880() {
                 "2%", "1%", "2%", "2%", "1%",
                 "0%", "0%", "1%", "1%", "1%",
                 "0", "2", "0.2", "0.6", "0.04",
-                "0.04", "0.000", "0.002",
+                "0.04", "0.001", "0.002",
```

Comment:
This test is checking that `.0005` rounded to 3 digits is "0.000" under HALF_EVEN.

This is wrong. See,
`new BigDecimal(0.0005) ==> 0.0005000000000000000104083408558608425664715468883514404296875`. 

Resolving the bug in this PR requires fixing this incorrect test case. From the JDK-4241880 issue, it seems this case may have been put in to "affirm the incorrect behavior".